src/overlays/actors/ovl_En_Kakasi3/z_en_kakasi3.c

```diff
@@ -21,35 +47,382 @@ const ActorInit En_Kakasi3_InitVars = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A915B8.s")
+        if (globalCtx->msgCtx.msgMode == 0) {
+            func_8010B680(globalCtx, 0x40A6, NULL);
+            this->dialogState = 5;
+            func_800803F0(globalCtx, this->camId);
+            this->camId = -1;
+            func_8002DF54(globalCtx, NULL, 8);
+            this->actionFunc = func_80A91A90;
+            return;
+        }
+    }
+    if ((globalCtx->msgCtx.unk_E3EE == 3) && (globalCtx->msgCtx.msgMode == 0)) {
+        globalCtx->msgCtx.unk_E3EE = 4;
+        if (BREG(3) != 0) {
+            osSyncPrintf("\n\n");
+            // With this, other guys are OK! That's it!
+            osSyncPrintf(VT_FGCOL(CYAN) "☆☆☆☆☆ これで、他の奴もＯＫ！だ！ ☆☆☆☆☆ %d\n" VT_RST,
+                         globalCtx->msgCtx.unk_E3EE);
+        }
+        this->unk_195 = 1;
+        func_8010B680(globalCtx, 0x40A7, NULL);
+        this->dialogState = 5;
+        func_8002DF54(globalCtx, NULL, 8);
+        this->actionFunc = func_80A91A90;
+    } else if (globalCtx->msgCtx.unk_E3EE == 1) {
+        func_80A90EBC(this, globalCtx, 0);
+        player->stateFlags2 |= 0x800000;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A91620.s")
+void func_80A91A90(EnKakasi3* this, GlobalContext* globalCtx) {
+    func_80A90E28(this);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_8002DF54(globalCtx, NULL, 8);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A91760.s")
+    if (this->dialogState == func_8010BDBC(&globalCtx->msgCtx) && func_80106BC8(globalCtx)) {
+        if (this->unk_195 != 0) {
+            if ((gSaveContext.eventChkInf[9] & 0x1000) == 0) {
+                gSaveContext.eventChkInf[9] |= 0x1000;
+            }
+        }
+        if (globalCtx->cameraPtrs[this->camId] == 0) {
+            this->camId = -1;
+        }
+        if (this->camId != -1) {
+            func_8005B1A4(globalCtx->cameraPtrs[this->camId]);
+        }
+        func_80106CCC(globalCtx);
+        globalCtx->msgCtx.unk_E3EE = 4U;
+        func_8002DF54(globalCtx, NULL, 7);
+        this->actionFunc = func_80A911F0;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A917FC.s")
+void EnKakasi3_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi3* this = THIS;
+    s32 pad;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A9187C.s")
+    if (BREG(2) != 0) {
+        osSyncPrintf("\n\n");
+        // flag!
+        osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ フラグ！ ☆☆☆☆☆ %d\n" VT_RST, gSaveContext.unk_12C5);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi3/func_80A918E4.s")
+    this->unk_198 += 1;
+    this->actor.posRot.rot = this->actor.shape.rot;
+    for (i = 0; i < 4; i++) {
+        if (this->unk_19C[i] != 0) {
+            this->unk_19C[i]--;
```

Comment:
You have `s16 unk_19C[0x3];` (notice the length, 3, vs this for loop assuming 4), could `unk_1A2` actually be part of that array instead of being on its own?